### ISSUE TITLE
rewrite units for plotting

### ISSUE THREAD

**Issue by dcherian** [xarray-contrib/cf-xarray#156]
It would be nice if we could rewrite the `units` attribute when using `.cf.plot()` so `units = "kg/m3"` becomes `units = "kg/m³"`. There must be some convenient way of doing this, maybe with pint?

cc @keewis @jthielen

**Comment by jthielen** [comment]
Yes, this could definitely be done with Pint (see the formatting options here: https://pint.readthedocs.io/en/stable/tutorial.html#string-formatting)! Though, in order to interpret CF/UDUNITS-style exponents, a custom preprocessor ([like that in MetPy](https://github.com/Unidata/MetPy/blob/3adfbd81629f0915f8d53e1e7b2509b2628827ca/src/metpy/units.py#L35-L39)) is needed, which connects back to the larger issue of  https://github.com/xarray-contrib/pint-xarray/issues/26.

**Comment by dcherian** [comment]
Thanks.

Another option is `cf_units`
``` python
import cf_units

for string in ["degreeN", "m2/s4", "kg/m3", "degC"]:
    print(cf_units.Unit(string).format(cf_units.UT_UTF8))
```

which looks good aside for the reversed order for `kg/m³`
```
°
m²·s⁻⁴
m⁻³·kg
°C
```

**Comment by aulemahal** [comment]
One argument against cf_units is that it is not pip-installable. It requires the UDUNITS-2 library which is not on pypi. Works well with conda, but potential headaches otherwise!
